kms/api-client/asymmetric.py

```diff
@@ -13,71 +13,100 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.rom googleapiclient import discovery
 
-import base64
 import hashlib
 
 from cryptography.exceptions import InvalidSignature
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import hashes, serialization
 from cryptography.hazmat.primitives.asymmetric import ec, padding, utils
 
+from google.cloud import kms_v1
+from google.cloud.kms_v1 import enums
+
+
+# [START kms_create_asymmetric_key]
+def create_asymmetric_key(project_id, location_id, key_ring_id, crypto_key_id):
+    """Creates an RSA encrypt/decrypt key pair within a specified KeyRing."""
+
+    # Creates an API client for the KMS API.
+    client = kms_v1.KeyManagementServiceClient()
+
+    # The resource name of the KeyRing associated with the CryptoKey.
+    parent = client.key_ring_path(project_id, location_id, key_ring_id)
+
+    # Create the CryptoKey object template
+    purpose = enums.CryptoKey.CryptoKeyPurpose.ASYMMETRIC_DECRYPT
+    algorithm = enums.CryptoKeyVersion.CryptoKeyVersionAlgorithm.\
+        RSA_DECRYPT_OAEP_2048_SHA256,
+    crypto_key = {'purpose': purpose,
+                  'version_template': {'algorithm': algorithm}}
+
+    # Create a CryptoKey for the given KeyRing.
+    response = client.create_crypto_key(parent, crypto_key_id, crypto_key)
+
+    print('Created CryptoKey {}.'.format(response.name))
+    return response
+# [END kms_create_asymmetric_key]
+
 
 # [START kms_get_asymmetric_public]
-def getAsymmetricPublicKey(client, key_path):
+def get_asymmetric_public_key(key_name):
     """
     Retrieves the public key from a saved asymmetric key pair on Cloud KMS
 
+    Example key_name:
+      "projects/PROJECT_ID/locations/global/keyRings/RING_ID/cryptoKeys/KEY_ID"
+
     Requires:
       cryptography.hazmat.backends.default_backend
       cryptography.hazmat.primitives.serialization
     """
-    request = client.projects() \
-                    .locations() \
-                    .keyRings() \
-                    .cryptoKeys() \
-                    .cryptoKeyVersions() \
-                    .getPublicKey(name=key_path)
-    response = request.execute()
-    key_txt = response['pem'].encode('ascii')
+
+    client = kms_v1.KeyManagementServiceClient()
+    response = client.get_public_key(key_name)
```

Comment:
This function requires the name of a CryptoKeyVersion, not just a CryptoKey, so it fails with names like the example key_name.

Reply:
good catch